frontend/src/pages/Home/HomePage.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import axios from 'axios';
+import Loading from './components/ProductList/Loading/Loading';
+import { ProductProps } from './components/ProductList/Product/Product';
+import { rootState } from '../../assets/interface/interface';
+import Criteria from './components/Criteria/Criteria';
+import ProductList from './components/ProductList/ProductList';
+import './HomePage.scss';
+
+const HomePage: React.FC = () => {
+  const [products, setProducts] = useState<ProductProps[]>([]);
```

Comment:
we can separate the logic for this component into a new hook. ex: `useProductLoad`